hw/dv/tools/dvsim/vcs.hjson

```diff
@@ -125,31 +125,45 @@
   cov_merge_dir:    "{scratch_path}/cov_merge"
   cov_merge_db_dir: "{cov_merge_dir}/merged.vdb"
   cov_merge_cmd:    "{job_prefix} urg"
-  cov_merge_opts:   ["-full64",
-                     "+urg+lic+wait",
-                     "-nocheck",
+  cov_merge_opts:   ["-lca",
+                     "-full64",
+                     // No need of generating report when merging coverage.
                      "-noreport",
-                     "-flex_merge drop",
-                     "-group merge_across_scopes",
+                     // Merge results from tests in parallel.
                      "-parallel",
                      "-parallel_split 20",
+                     "-parallel_temproot {cov_merge_dir}",
+                     "+urg+lic+wait",
+                     // Merge same assert instances found in different VDBs.
+                     "-merge_across_libs",
+                     // Enable union mode of flexible merging for covergroups.
+                     "-flex_merge union",
                      // Use cov_db_dirs var for dir args; append -dir in front of each
                      "{eval_cmd} echo {cov_db_dirs} | sed -E 's/(\\S+)/-dir \\1/g'",
                      "-dbname {cov_merge_db_dir}"]
 
   // Generate coverage reports in text as well as html.
-  cov_report_dir:       "{scratch_path}/cov_report"
-  cov_report_cmd:       "{job_prefix} urg"
-  cov_report_opts:      ["-full64",
-                        "+urg+lic+wait",
-                        "-dir {cov_merge_db_dir}",
-                        "-group instcov_for_score",
-                        "-line nocasedef",
-                        "-format both",
-                        "-elfile {vcs_cov_excl_files}",
-                        "-report {cov_report_dir}"]
-  cov_report_txt:       "{cov_report_dir}/dashboard.txt"
-  cov_report_page:      "dashboard.html"
+  cov_report_dir:   "{scratch_path}/cov_report"
+  cov_report_cmd:   "{job_prefix} urg"
+  cov_report_opts:  ["-lca",
```

Comment:
There is also cost-based grading, which can be handy. For the grader I wrote a few years ago I used "-grade cost clocktime", so tests that achieve the highest coverage in the least wall-clock time are preferred. I also added "testfile" after "clocktime", per the urg manual:
"To generate the gradedtests.txt test file, specify the testfile argument with the –grade option..."

I also recall grading could take lots of time, so we had to figure out which set of options was preferable. I can do an archaeological dive if this becomes an issue.

Reply:
Agree - lets fine tune in multiple passes.